rootfs/apply.go

```diff
@@ -69,7 +69,7 @@ func ApplyLayer(snapshots snapshot.Snapshotter, mounter Mounter, rd io.Reader, p
 	digester := digest.Canonical.Digester() // used to calculate diffID.
 	rd = io.TeeReader(rd, digester.Hash())
 
-	if _, err := archive.Apply(context.Background(), key, rd); err != nil {
+	if _, err := archive.Apply(context.Background(), dir, rd); err != nil {
```

Comment:
Thanks @dmcgowan for review, but I have a different view. 
The function of `archive.Apply` is to read the image layer, and stored into a specified directory.
So, the parameter should be a  `directory path` like the `dir` in `mounter.Mount(dir, mounts...)`, but not be a `key`.